src/CustomHeaderDataFetcher.ts

```diff
@@ -0,0 +1,24 @@
+import DataFetcher, {
+  DataFetcherConfig,
+} from "@thepalaceproject/web-opds-client/lib/DataFetcher";
+
+export interface CustomHeaderDataFetcherConfig extends DataFetcherConfig {
+  extraHeaders?: Record<string, string>;
+}
+
+export default class CustomHeaderDataFetcher extends DataFetcher {
+  private readonly extraHeaders: Record<string, string>;
+
+  constructor({
+    extraHeaders = {},
+    ...dataFetcherConfig
+  }: CustomHeaderDataFetcherConfig = {}) {
+    super(dataFetcherConfig);
+    this.extraHeaders = extraHeaders;
+  }
+
+  fetch(url: string, { headers = {}, ...restOptions } = {}) {
+    const customizedHeaders = Object.assign({}, this.extraHeaders, headers);
+    return super.fetch(url, { headers: customizedHeaders, ...restOptions });
```

Comment:
I'm not 100% on this, but I was wondering if we should modify this to be `NoCacheDataFetcher`, and make sure we are bypassing the browser cache as well by setting the `cache: no-cache` option to fetch as well.

I'm not totally clear if we also need to set the headers when setting the cache option, or if that option will set the headers as well.

Reply:
Sure. We can do that. I'm happy to make those changes, if you like. Otherwise, feel free to take this PR wherever you want. This was mainly to see what would be necessary. And it provides a bit of roadmap of how to do it and type it without pulling in much from the superclass.